benches/bench.rs

```diff
@@ -25,7 +27,41 @@ mod poly_benches {
                 };
                 b.iter_with_setup(rand_factors, |(lhs, rhs)| &lhs * &rhs)
             },
-            &[5, 10, 20, 40],
+            &TEST_DEGREES,
+        );
+    }
+
+    // Benchmarks subtraction of two polynomials
+    fn subtraction(c: &mut Criterion) {
+        let mut rng = rand::thread_rng();
```

Comment:
We're instantiating a new RNG each time we run the test; could we maybe use a fixed seed RNG? If the number of random values used is large enough, it would not be an issue to use a new instance each time, but neither would reusing the seed be then.

Just a thought. One could go for the whole "traditional" `<R: rand::Rng>(rng: R, ...` setup, but I don't think that `criterion` makes it easy to work with this stuff.

Reply:
Hm, I thought thread local rngs were cached? I think we're only creating one per function and then it is re-used for each input. Or do you mean using the same one for all the benchmarks by instantiating it in the pre-test setup?

Reply:
The result of `thread_rng` depends on a seed provided by the system. The seed is something we would like to control for reproducibility purposes. You could provide a constant seed instead.

Reply:
Ah, yeah, if the goal is reproducibility, that makes sense. Normally, the rng will re-seed itself automatically after a certain number of bytes are produced, so we'd need to disable that as well. I'll look through the rng docs and see what the options are.

Reply:
So it looks like we can do this:
```rust
const RNG_SEED: [usize; 1] = [42];
use rand::SeedableRng;
let mut rng = rand::StdRng::from_seed(&RNG_SEED);
```
(I didn't include the code that wouldn't change)
What do you think of that?

Reply:
Maybe even `rand::XorShiftRng` which is the fastest seedable PRNG in `rand`?

Reply:
OK, I reworked all of them to use XorShiftRng. One thing to note is that we're at 0.4.2 of rand, and the latest is 0.5.5. It looks like there were quite a few changes between the versions, so this may need to be refactored if we upgrade the rand crate version in the future.